cmd/cert-checker/main.go

```diff
@@ -170,12 +175,18 @@ func (c *certChecker) checkCert(cert core.Certificate) (problems []string) {
 		if parsedCert.IsCA {
 			problems = append(problems, "Certificate can sign other certificates")
 		}
-		// Check the cert has the correct validity period +/- an hour
-		if parsedCert.NotAfter.Sub(cert.Issued) > checkPeriod+time.Hour {
-			problems = append(problems, "Certificate has a validity period longer than 90 days")
-		} else if parsedCert.NotAfter.Sub(cert.Issued) < checkPeriod-time.Hour {
-			problems = append(problems, "Certificate has a validity period shorter than 90 days")
+		// Check the cert has the correct validity period
+		validityPeriod := parsedCert.NotAfter.Sub(parsedCert.NotBefore)
+		if validityPeriod > checkPeriod {
+			problems = append(problems, fmt.Sprintf("Certificate has a validity period longer than %s", checkPeriod))
+		} else if validityPeriod < checkPeriod {
+			problems = append(problems, fmt.Sprintf("Certificate has a validity period shorter than %s", checkPeriod))
```

Comment:
Being this exact makes me nervous about having `checkPeriod` be a compiled-in constant.  Please file a follow-on issue to have the checker read a boulder config file and use the validity period specified in `c.CA.CFSSL.Signing.Profiles[ca.CA.Profile].Expiry`.